_opensearch/query-dsl/index.md

```diff
@@ -120,3 +120,36 @@ With query DSL, however, you can include an HTTP request body to look for result
 }
 ```
 The OpenSearch query DSL comes in three varieties: term-level queries, full-text queries, and boolean queries. You can even perform more complicated searches by using different elements from each variety to find whatever data you need.
+
+## A note on special characters in field values 
+
+Due to word boundaries associated with Unicode special characters, the Unicode standard analyzer cannot index field values as whole values when they contain these special characters. As a result, a field value that includes a special character is parsed by the standard analyzer as multiple values separated by the special character, effectively tokenizing the different elements either side of it. This can lead to unintentional requests and search query outcomes, including those that can have an impact on security. Therefore, special characters should be avoided in field values handled by query DSL unless a custom analyzer is used. 
+
+The examples below illustrate values containing special characters that will be parsed improperly by the standard analyzer. In this example, the existence of the hyphen/minus sign in the value prevents the analyzer from distinguishing between the two different users for `user.id` and interprets them as one and the same:
+
+```json
+{
+  "bool": {
+    "must": {
+      "match": {
+        "user.id": "User-1"
+      }
+    }
+  }
+}
+```
+
+```json
+{
+  "bool": {
+    "must": {
+      "match": {
+        "user.id": "User-2"
```

Comment:
Rather than providing only negative examples, what about providing a 'do this instead of that' formula?  For this case specifically using `keyword` field is the more accurate mechanism [1] [2]


[1] https://opensearch.org/docs/2.2/opensearch/supported-field-types/keyword/
[2] https://opster.com/guides/elasticsearch/search-apis/elasticsearch-strings-keyword-vs-text-vs-wildcard/

Reply:
@peternied That provides an opportunity to better clarify the problem, too. I edited the note based on comments above. See if that comes across better. Thanks.